realm-elasticsearch/spec/spec_helper.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module SpecHelpers
+  def wait_for(timeout = 30, increment = 0.2)
+    Timeout.timeout(timeout) do
+      loop do
+        result = yield
+        return result if result
```

Comment:
infinite loop posible. I suggest to think about `break` condition for safety ;-)

Reply:
It's actually the point on this code. If there is async code that never fulfils the condition in test it will be break with timeout exception.